applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -0,0 +1,856 @@
+{
+    // Note that the rhs has double the size
+    if (rRightHandSideVector.size() != 2 * NumNodes)
+        rRightHandSideVector.resize(2 * NumNodes, false);
+    rRightHandSideVector.clear();
+
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    const auto& r_geometry = this->GetGeometry();
+    GeometryUtils::CalculateGeometryData(r_geometry, data.DN_DX, data.N, data.vol);
+    GetWakeDistances(data.distances);
+
+    const double density = ComputeDensity(rCurrentProcessInfo);
+
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+    array_1d<double, Dim> upper_velocity = PotentialFlowUtilities::ComputeVelocityUpperWakeElement<Dim,NumNodes>(*this);
+    array_1d<double, Dim> lower_velocity = PotentialFlowUtilities::ComputeVelocityLowerWakeElement<Dim,NumNodes>(*this);
+
+    for (unsigned int i = 0; i < Dim; i++){
+        upper_velocity[i] += free_stream_velocity[i];
+        lower_velocity[i] += free_stream_velocity[i];
+    }
+    const array_1d<double, Dim> diff_velocity = upper_velocity - lower_velocity;
+
+    const BoundedVector<double, NumNodes> upper_rhs = - data.vol * density * prod(data.DN_DX, upper_velocity);
+    const BoundedVector<double, NumNodes> lower_rhs = - data.vol * density * prod(data.DN_DX, lower_velocity);
+    const BoundedVector<double, NumNodes> wake_rhs = - data.vol * density * prod(data.DN_DX, diff_velocity);
+
+    if (this->Is(STRUCTURE)){
+        double upper_vol = 0.0;
+        double lower_vol = 0.0;
+
+        CalculateVolumesSubdividedElement(upper_vol, lower_vol, rCurrentProcessInfo);
+        for (unsigned int i = 0; i < NumNodes; ++i){
+            if (r_geometry[i].GetValue(TRAILING_EDGE)){
+                rRightHandSideVector[i] = upper_rhs(i)*upper_vol/data.vol;
+                rRightHandSideVector[i + NumNodes] = lower_rhs(i)*lower_vol/data.vol;
+            }
+            else{
+                AssignRightHandSideWakeNode(rRightHandSideVector, upper_rhs, lower_rhs, wake_rhs, data, i);
+            }
+        }
+    }
+    else{
+        for (unsigned int i = 0; i < NumNodes; ++i){
+            AssignRightHandSideWakeNode(rRightHandSideVector, upper_rhs, lower_rhs, wake_rhs, data, i);
+        }
+    }
+}
+
+template <int Dim, int NumNodes>
+void TransonicPerturbationPotentialFlowElement<Dim, NumNodes>::CalculateLeftHandSideSubdividedElement(
+    Matrix& lhs_positive,
+    Matrix& lhs_negative,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    GetWakeDistances(data.distances);
+
+    // Subdivide the element
+    constexpr unsigned int nvolumes = 3 * (Dim - 1);
+    BoundedMatrix<double, NumNodes, Dim> Points;
+    array_1d<double, nvolumes> PartitionsSign;
+    BoundedMatrix<double, nvolumes, NumNodes> GPShapeFunctionValues;
```

Comment:
```suggestion
    BoundedMatrix<double, nvolumes, NumNodes> GPShapeFunctionValues;
```
Local variables should be named with `snake_case` (not `CamelCase`).